src/checker/zx/ZXChecker.cpp

```diff
@@ -0,0 +1,147 @@
+#include "checker/zx/ZXChecker.hpp"
+
+#include "Definitions.hpp"
+#include "QuantumComputation.hpp"
+#include "Simplify.hpp"
+#include "ZXDiagram.hpp"
+#include "dd/Definitions.hpp"
+#include "zx/FunctionalityConstruction.hpp"
+
+#include <chrono>
+#include <cstddef>
+#include <iostream>
+#include <numeric>
+#include <optional>
+#include <unordered_set>
+
+namespace ec {
+    ZXEquivalenceChecker::ZXEquivalenceChecker(const qc::QuantumComputation& qc1, const qc::QuantumComputation& qc2, Configuration configuration) noexcept:
+        EquivalenceChecker(qc1, qc2, std::move(configuration)), miter(zx::FunctionalityConstruction::buildFunctionality(&qc1)), tolerance(configuration.functionality.traceThreshold) {
```

Comment:
here, a different default is used for the `tolerance` as opposed to the header file.

Reply:
Correct, I guess the proper way would be to default to the one in the header file in case the traceThreshold is 0.0. Does that happen?

Reply:
I guess that would be the proper way, yes.

Shouldn't really happen, but ... who knows what people come up with.

Reply:
A traceThreshold of 0.0 means no distinct phases are considered equal no matter how close. I guess that could be the intention.

Reply:
Yeah. Sounds reasonable. Although I would not recommend setting it to that value 🙃